### DISCUSSION TITLE
Zewnętrzna antena Wifi, jakieś pomysły?

### DISCUSSION THREAD

**Discussion by tnowak** [hackerspace-silesia/Smogomierz#127] [General]
Hej,

Okazuje się, że mam na balkonie dość słaby zasięg Wifi. Tzn. sięga, ale w moim bloku jest b. dużo AP na 2,4GHz i jest śmietnik w eterze w tej częstotliwości. Tak czy siak RSSI oscyluję mi w okolicach -70 dBm i zdarzają się nawet sytuacje, w których urządzenie jest niedostępne.

Wstępnie rozważam dwie opcje: dołożyć Unifi mesha na balkonie, ale to trochę overkill, albo... no właśnie i tu pytanie. Czy ktoś dokładał zewnętrzną antenę do ESP na NodeMCU v3? Da się? Duża rzeźba?

**Comment by bfaliszek** [comment]
ESP8266 ESP-07 ma złącze antenowe U.FL.

**Comment by tnowak** [comment]
Uhm. Czyli co, musiałbym wylutować z NodeMCU v3 to co tam jest i wlutować taki ESP-07 tak?

**Comment by tnowak** [comment]
To jeszcze jedno pytanie. Jakbym kupił Wemos D1 Mini Pro taki jak tutaj:
https://allegro.pl/oferta/wemos-d1-mini-pro-16m-v1-0-8309943224
to smogly na nim bez problemu ruszy, @bfaliszek ?

**Comment by bfaliszek** [comment]
Powinien działać bez problemu, tylko pewnie będziesz musiał samemu skompilować kod, bo ten który jest dla esp8266 jest dla wersji 4mb

**Comment by bfaliszek** [comment]
> Uhm. Czyli co, musiałbym wylutować z NodeMCU v3 to co tam jest i wlutować taki ESP-07 tak?

Nie, takie coś nie zadziała. ESP8266 mają różne wersja. NodeMCU używa ESP-12. Jest też np. ESP-01, które ma tylko 2 piny gpio. ESP-07 ma za to zewnętrzną antenę. Każde z nich to esp8266 i będzie na nich działał kod smogomierza.

**Comment by tnowak** [comment]
Ok, czyli realnie zostaje mi tylko opcja tego D1 Mini Pro 16m v1.0 i samodzielnie skomplikowanie kodu. Znalazłem już instrukcję na to, ale muszę zważyć czy warto się w to pakować, bo tę kompilację musiałbym robić wielokrotnie aby utrzymać aktualność softu z dystrybucją. Jeszcze rozważam drastyczną wersję, tj. zrobienie przerwy na antenie na ESP8266 na NodeMCU i przylutowanie złączy anteny tam - widzę w necie, że ludzie też tak robią...
